[docs] Document `on.ref_delete` workflow trigger (#46913)
expo/expo: docs/pages/eas/workflows/syntax.mdx

```

### `on.ref_delete`

Runs your workflow when a GitHub branch or tag is deleted. Requires a [linked GitHub repository](/eas/workflows/get-started/#automate-workflows-with-github-events) on your EAS project.

> With the `branches` list, you can trigger the workflow only when those specified branches are deleted. For example, if you use `branches: ['main']`, only deletion of the `main` branch triggers the workflow. Supports globs. By using the `!` prefix you can specify branches to ignore (you still need to provide at least one branch pattern without it). See [`on.push`](#onpush) for details on glob and negation pattern matching.

Reviewer: i like you excluded `main` in the code example, maybe we can use a different example here too (not `main`)?

With the `tags` list, you can trigger the workflow only when those specified tags are deleted. Supports globs and `!` negation with the same rules as branches.

When neither `branches` nor `tags` are provided, `branches` defaults to `['*']` and `tags` defaults to `[]`, which means the workflow triggers when any branch is deleted and does not trigger on tag deletions. If only one of the two lists is provided the other defaults to `[]`.

> **info** Workflow files are read from the default branch HEAD at the time of deletion, not from the deleted ref.

```yaml
on:
  # @info #
  ref_delete:
    # @end #
    branches:
      - feat/**
      - !main # other branch names and globs

> 

Reviewer: ```suggestion
```

Author: `pnpm format` does this, otherwise lint fails

```

### `on.ref_delete`

Runs your workflow when a GitHub branch or tag is deleted. Requires a [linked GitHub repository](/eas/workflows/get-started/#automate-workflows-with-github-events) on your EAS project.

> With the `branches` list, you can trigger the workflow only when those specified branches are deleted. For example, if you use `branches: ['main']`, only deletion of the `main` branch will trigger the workflow. Supports globs. By using the `!` prefix you can specify branches to ignore (you still need to provide at least one branch pattern without it). See [`on.push`](#onpush) for details on glob and negation pattern matching.

Reviewer 2: **[nit]** Prefer present tense over "will" for system behavior (`style-guide.md#voice-and-tone`). `on.push` and `on.pull_request` use the same phrasing, so you may prefer a consistent pass instead.

```suggestion
With the `branches` list, you can trigger the workflow only when those specified branches are deleted. For example, if you use `branches: ['main']`, only deletion of the `main` branch triggers the workflow. Supports globs. By using the `!` prefix you can specify branches to ignore (you still need to provide at least one branch pattern without it). See [`on.push`](#onpush) for details on glob and negation pattern matching.
```

Author: Corrected in https://github.com/expo/expo/pull/46913/commits/daa0852c3e12b37022704e82247b510cc6619b36

With the `tags` list, you can trigger the workflow only when those specified tags are deleted. Supports globs and `!` negation with the same rules as branches.

> When neither `branches` nor `tags` are provided, `branches` defaults to `['*']` and `tags` defaults to `[]`, which means the workflow will trigger when any branch is deleted and will not trigger on tag deletions. If only one of the two lists is provided the other defaults to `[]`.

Reviewer 2: **[nit]** Same present-tense fix (`style-guide.md#voice-and-tone`).

```suggestion
When neither `branches` nor `tags` are provided, `branches` defaults to `['*']` and `tags` defaults to `[]`, which means the workflow triggers when any branch is deleted and does not trigger on tag deletions. If only one of the two lists is provided the other defaults to `[]`.
```

Author: Corrected in https://github.com/expo/expo/pull/46913/commits/daa0852c3e12b37022704e82247b510cc6619b36
